libcrux-iot-testutil/src/events.rs

```diff
@@ -0,0 +1,235 @@
+extern crate alloc;
+
+use alloc::string::{String, ToString};
+
+/// The top-level event
+#[derive(Debug, PartialEq, Eq)]
+pub enum TestUtilEvent {
+    /// This is emitted once at the start of a test suite
+    Launch(LaunchEvent),
+    /// These are events related to tests
+    Test(TestEvent),
+    /// These are events related to benchmarks
+    Benchmark(BenchmarkEvent),
+    /// These are user-generated events, most likely for logging
+    User(String),
+}
+
+/// This is emitted once at the start of a test suite
+#[derive(Debug, PartialEq, Eq)]
+pub struct LaunchEvent {
+    /// The name of the launched test suite.
+    pub name: String,
+    /// The frequency of the CPU core.
+    pub core_freq: u32,
+}
+
+/// These are events related to tests
+#[derive(Debug, PartialEq, Eq)]
+pub enum TestEvent {
+    /// This is emitted right before a test is run
+    Run { name: String },
+    /// This is emitted if the runner encounters a test that should be skipped
+    Skip { name: String },
+    /// This is emitted if the test ran successfully
+    Pass { name: String },
+    /// This is emitted if the test failed
+    Error { name: String, error_message: String },
+}
+
+/// These are events related to benchmarks
+#[derive(Debug, PartialEq, Eq)]
+pub enum BenchmarkEvent {
+    /// This is emitted when an iteration of a benchmark is started
+    Run { name: String, run_id: u32 },
+    /// This is emitted if the runner encounters a test that should be skipped
+    Skip { name: String },
+    /// This is emitted when an iteration of a benchmark is finished
+    Done {
+        name: String,
+        run_id: u32,
+        cycles: u32,
+    },
+    /// This is emitted when an iteration of a benchmark returned an error
+    Error {
+        name: String,
+        run_id: u32,
+        error_message: String,
+    },
+}
+
+impl TestUtilEvent {
```

Comment:
We might, at a later point, want to have some independent documentation for the custom format once it's sufficiently stable.